src/access/acp.test.ts

```diff
@@ -3961,3 +4064,2232 @@ describe("getAgentAccessAll", () => {
+      expect(policyUrls).toContain("https://some.pod/resource?ext=acr#policy");
+      const updatedAcr = internal_getAcr(updatedResourceWithAcr!);
+      const existingPolicy = getPolicy(
+        updatedAcr,
+        "https://some.pod/resource?ext=acr#policy"
+      )!;
+      expect(getIriAll(existingPolicy, acp.allOf)).toStrictEqual([
+        "https://some.pod/resource?ext=acr#own-rule",
+      ]);
+      expect(getIriAll(existingPolicy, acp.noneOf)).toStrictEqual([
+        "https://some.pod/resource?ext=acr#other-rule",
+      ]);
+      const existingRule = getRule(
+        updatedAcr,
+        "https://some.pod/resource?ext=acr#own-rule"
+      )!;
+      expect(getIri(existingRule, acp.agent)).toBe(webId);
+    });
+
+    it("does not affect other actor's access", () => {
+      const otherWebId = "https://arbitrary-other.pod/profile#other-actor";
+      const resourceWithAcr = mockResourceWithAcr(
+        "https://some.pod/resource",
+        "https://some.pod/resource?ext=acr",
+        {
+          policies: {
+            "https://some.pod/resource?ext=acr#policy": {
+              allOf: {
+                "https://some.pod/resource?ext=acr#applicable-allOf-rule": {
+                  [acp.agent]: [otherWebId],
+                },
+              },
+              deny: {
+                read: true,
+              },
+            },
+          },
+          memberPolicies: {},
+          acrPolicies: {},
+          memberAcrPolicies: {},
+        }
+      );
+
+      const updatedResourceWithAcr = internal_setActorAccess(
+        resourceWithAcr,
+        acp.agent,
+        webId,
+        {
+          read: true,
+        }
+      );
+
+      expect(
+        internal_getAgentAccess(updatedResourceWithAcr!, otherWebId)
+      ).toStrictEqual({
+        read: false,
+      });
+    });
+
+    it("does not remove existing Policies that no longer apply to this Resource, but might still apply to others", () => {
+      const resourceWithAcr = mockResourceWithAcr(
+        "https://some.pod/resource",
+        "https://some.pod/resource?ext=acr",
+        {
+          policies: {
+            "https://some.pod/resource?ext=acr#policy": {
+              deny: { read: true },
+              allOf: {
+                "https://some.pod/resource?ext=acr#rule": {
+                  [acp.agent]: [webId],
+                },
+              },
+            },
+          },
+          memberPolicies: {},
+          acrPolicies: {},
+          memberAcrPolicies: {},
+        }
+      );
+
+      const updatedResourceWithAcr = internal_setActorAccess(
+        resourceWithAcr,
+        acp.agent,
+        webId,
+        {
+          read: true,
+        }
+      );
+
+      const acr = internal_getAcr(updatedResourceWithAcr!);
+      expect(
+        getThing(acr, "https://some.pod/resource?ext=acr#policy")
+      ).not.toBeNull();
+    });
+
+    it("does not remove references to Policies that do not exist in this ACR", () => {
```

Comment:
"Policies that do not exist in this ACR" would be equivalent to "empty Policies", right ? I don't think it's  a more correct phrasing, I'm just making sure I get it right :)

Reply:
It's kinda arbitrary, but I distinguished between "Policies that do not exist" meaning that there are no Triples with the given Policy URL as a Subject, and "empty Policies" meaning Policies without Rules and allow/deny modes (but they could have other Triples, e.g. `rdf:type acp:Policy`).